src/components/input/radio/index.tsx

```diff
@@ -0,0 +1,76 @@
+import {
+	Component,
+	ComponentWillLoad,
+	Event,
+	EventEmitter,
+	h,
+	Host,
+	Listen,
+	Method,
+	Prop,
+	VNode,
+	Watch,
+} from "@stencil/core"
+import { Color, Data } from "../../../model"
+import { Clearable } from "../Clearable"
+import { Input } from "../Input"
+import { Looks } from "../Looks"
+import { Selectable } from "./Selected"
+
+@Component({
+	tag: "smoothly-input-radio",
+	styleUrl: "style.css",
+	scoped: true,
+})
+export class SmoothlyInputRadio implements Input, Clearable, ComponentWillLoad {
+	active?: Selectable
+	@Prop({ mutable: true }) value: any = undefined
+	@Prop({ mutable: true, reflect: true }) looks: Looks = "plain"
+	@Prop() clearable?: boolean
+	@Prop() name: string
+	@Event() smoothlyInputLooks: EventEmitter<(looks: Looks, color: Color) => void>
+	@Event() smoothlyInput: EventEmitter<Data>
+	@Event() smoothlyInputLoad: EventEmitter<(parent: HTMLElement) => void>
+	componentWillLoad(): void | Promise<void> {
+		this.smoothlyInputLooks.emit(looks => (this.looks = looks))
+		this.smoothlyInput.emit({ [this.name]: this.value })
+		this.smoothlyInputLoad.emit(() => {
+			return
+		})
+	}
+	@Listen("smoothlyRadioButtonRegister")
+	handleRegister(event: CustomEvent<(name: string) => void>) {
+		event.detail(this.name)
+		event.stopPropagation()
+	}
+	@Listen("smoothlySelect")
+	smoothlyRadioInputHandler(event: CustomEvent<Selectable>): void {
+		if (this.clearable && this.active?.value === event.detail.value) {
```

Comment:
non needed braces

Reply:
braces are needed! on next row this.clear() is run and then this.value = undefined, and that satisfies the next if check so that will be run but it shouldnt

Reply:
Apparently braces could mean many things depending on where you source it :upside_down_face:
I have always called `{}` braces :upside_down_face:
I am assuming you thought I meant parentheses `()` [If there is only one thing done the line after an if statement you do not need `{}` around said if block](https://www.typescriptlang.org/play?#code/MYewdgzgLgBLC8MBEALApgGwyJAoXAlgGYwAUsAZBXAHTQCGATlBAOoFQqmqYZICU-XDBExQkEBjQ1sAc27RGBMLJgEIcRgFdOATxj0wAExgNmGgO4cUMAOTostgbkwQ0MAN7DR4iJOlyClBKKmoaRPQYELrOomLgflIyIPJIoFoYJgC2aIYwWYb6nMqyEM4AvkA). We typically leave it out unless we need it (more than one statement inside the block)

Reply:
I thought removing braces would force me to make the "else if" into an "if", thats why i refused at first, but i was wrong!